inventory.py

```diff
@@ -34,12 +37,17 @@
 SUBSCRIPTION_NAME = ENV['API_SUBSCRIPTION_NAME']
 API_SCOPE_PREFIX = ENV['API_SCOPE_PREFIX']
 MAX_REQ_ATTEMPTS = ENV['MAX_REQ_ATTEMPTS']
+CANVAS_TOKEN = ENV['CANVAS_TOKEN']
+CANVAS_URL = ENV['CANVAS_URL']
 
 UDW_CONN = psycopg2.connect(**ENV['UDW'])
 WAREHOUSE_INCREMENT = ENV['WAREHOUSE_INCREMENT']
 
 CREATE_CSVS = ENV.get('CREATE_CSVS', False)
 INVENTORY_DB = ENV['INVENTORY_DB']
+published_course_date = {}
+published_course_next_page_list = []
```

Comment:
This holds the pagination URL to query next for courses that don't get published_date from the first 100 results. So far only 2 courses from 5k published courses list I had to loop through

Reply:
Since library controls when to fetch the next results I need to hold it somewhere outside of the function call routine

Reply:
As we talked about over BlueJeans, I would rather see these defined within the function scope of some wrapper function around the recursive calls, or you could create a class and have these be instance variables and your functions can be methods. Using global variables and functions that don't return anything (but are really modifying these globals) seems confusing (at least to me) and probably hard to maintain. My two cents.

Reply:
@ssciolla I want to do it as separate PR since this approach is working and when this gets merged so that Tableau can pull directly from DB instead of csv's as it already started doing it. I might play around it a bit with refactoring. I don't know if I can completely avoid both the globals.